resources/certificate.rb

```diff
@@ -0,0 +1,30 @@
+#
+# Author:: Blair Hamilton (<blairham@me.com>)
+# Cookbook Name:: windows
+# Provider:: cert
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+actions :add, :delete
+
+attribute :store_name, :kind_of => String, :name_attribute => true
+attribute :cert_id, :kind_of => String
+attribute :infile, :kind_of => String
```

Comment:
Now I am not so sure. There isn't really a good unique element to use as the name. The store name is used in both adding and deleting where as the infile is just used in adding and cert_id is used in just deleting. 

The main problem I have with using store name as the name attribute I when adding multiple certificates to a store. I have a recipe where I am adding 3 certificates the TrustedPublisher store. In that case couldn't use the name attribute anyway since it would cause resource cloning. 

Adding certificates with `store_name` as `name_attribute`

    windows_cerificate 'TrustedPublisher 1' do
        action :add
        store_name 'TrustedPublisher'
        infile 'mycert.cer'
    end

    windows_cerificate 'TrustedPublisher 2' do
        action :add
        store_name 'TrustedPublisher'
        infile 'mycert2.cer'
    end

Adding certificates with `infile` as `name_attribute`

    windows_cerificate 'mycert.cer' do
        action :add
        store_name 'TrustedPublisher'
    end

    windows_cerificate 'mycert2.cer' do
        action :add
        store_name 'TrustedPublisher'
    end

Deleting certificates with `store_name` as `name_attribute`

    windows_cerificate 'TrustedPublisher 1' do
        action :delete
        store_name 'TrustedPublisher'
        cert_id '123456789'
    end

    windows_cerificate 'TrustedPublisher 2' do
        action :delete
        store_name 'TrustedPublisher'
        cert_id '123456789'
    end

Deleting certificates with `infile` as `name_attribute`

    windows_cerificate 'TrustedPublisher 1' do
        action :delete
        store_name 'TrustedPublisher'
        cert_id '123456789'
    end

    windows_cerificate 'TrustedPublisher 2' do
        action :delete
        store_name 'TrustedPublisher'
        cert_id '123456789'
    end